packages/kbn-alerts-grouping/src/components/alerts_grouping.tsx

```diff
@@ -118,6 +120,21 @@ const AlertsGroupingInternal = <T extends BaseAlertsGroupAggregations>(
     onOptionsChange,
   });
 
+  useEffect(() => {
```

Comment:
How about moving the initialization logic to the [grouping context provider](https://github.com/elastic/kibana/blob/b6287708f687d4e3288851052c0c6ae4ade8ce60/packages/kbn-alerts-grouping/src/contexts/alerts_grouping_context.tsx#L29-L28)?

```ts
export const AlertsGroupingContextProvider = ({
  initialState = {},
  children,
}: PropsWithChildren<{ initialState?: AlertsGroupingState }>) => {
  const [groupingState, setGroupingState] = useState<AlertsGroupingState>(initialState);
  return (
    <AlertsGroupingContext.Provider
      value={useMemo(
        () => ({ groupingState, setGroupingState }),
        [groupingState, setGroupingState]
      )}
    >
      {children}
    </AlertsGroupingContext.Provider>
  );
};
```

This way we could avoid the initialization effect